### PR TITLE
Screenshot scraping

### PR DESCRIPTION
**Pull request by Santabot123**
I added the option to extract text from screenshots. 
I also made this [Google Colab](https://colab.research.google.com/drive/1bRxx3km1YMWPi7P4HM4Ebf0I-zphagM7?usp=sharing) example of how to use it.  But I still recommend using this code locally, because the `select_area_with_opencv()` function allows you to select the area of the screenshot with your mouse, which makes it much more convenient to use.

### CODE REVIEW

**Comment by VinciGit00** [comment]
btw if you can also make a short video demo where you use the tool it could be useful

File: scrapegraphai/screenshot_scraping/screenshot_preparation.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        disabled=False
+    )
+
+    interact(update_plot, top_bottom=top_bottom_slider,
+             left_right=left_right_slider, image_size=image_size_bt)
+    
+    return left_right_slider, top_bottom_slider
+
+
+def crop_image(image, LEFT=None, TOP=None, RIGHT=None, BOTTOM=None,  save_path: str = None):
```
Comment (DiTo97) [inline scrapegraphai/screenshot_scraping/screenshot_preparation.py:179]: LGTM, why the uppercase arguments?

**Comment by Santabot123** [comment]
> btw if you can also make a short video demo where you use the tool it could be useful

I don't have a lot of experience with video editing, but I did my best. Here is the [link](https://youtu.be/2gU6fuQmIKE?si=InLNB2FX8USjH5l8) to youtube.

**Comment by DiTo97** [comment]
> > btw if you can also make a short video demo where you use the tool it could be useful
> 
> I don't have a lot of experience with video editing, but I did my best. Here is the [link](https://youtu.be/2gU6fuQmIKE?si=InLNB2FX8USjH5l8) to youtube.

appreciate the effort!

**Comment by VinciGit00** [comment]
hi thank you, in the next days we will make a blog post, would you like to be tagged?

**Comment by Santabot123** [comment]
> hi thank you, in the next days we will make a blog post, would you like to be tagged?

Yes

**Comment by VinciGit00** [comment]
ok, please give me your LinkedIn account. 
Im starting to use your script but I have this error, can you give me a tip?
Screenshot 2024-08-30 alle 12 12 47

**Comment by Santabot123** [comment]
I don't have a LinkedIn account. I thought you would just write something like “Thank you very much, [Santabot123](https://github.com/Santabot123) 😎”

**Comment by DiTo97** [comment]
yeah, @VinciGit00, maybe just link his GH account

**Comment by VinciGit00** [comment]
Ok I will

File: scrapegraphai/screenshot_scraping/text_detection.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from surya.ocr import run_ocr
```
Comment (DiTo97) [inline scrapegraphai/screenshot_scraping/text_detection.py:1]: @VinciGit00, try and place up top:
```python
import typing_extensions
```

**Pull request merged**